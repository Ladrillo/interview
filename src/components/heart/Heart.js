import React, { Component } from 'react';
import styled from 'styled-components';


export default function Heart({ on, toggle }) {
    return (
        <Styled on={on} onClick={toggle}>
            &#9825;
        </Styled>
    );
}

const Styled = styled.span`
    font-size: 16px;
    color: ${props => props.on ? '#FDF253' : '#222'};
    cursor: pointer;
    transform: scale(1);
    transition: all 0.4s 0.02s ease-in-out;

    &:hover {
        color: #FDF253;
        transform: scale(1.2);
        transition: all 0.4s 0.02s ease-in-out;
    }
`;