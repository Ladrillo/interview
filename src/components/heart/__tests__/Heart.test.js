import Heart from '../Heart';
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';


describe('Heart', () => {
    it('renders correctly', () => {
        const rendered = renderer.create(
            <Heart
                on     = {true}
                toggle = {()=>console.log('toggling...')}
            />
        );

        expect(rendered.toJSON()).toMatchSnapshot();
    });
});
