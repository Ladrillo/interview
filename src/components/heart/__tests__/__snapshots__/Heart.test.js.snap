exports[`Heart renders correctly 1`] = `
<span
  className="sc-bdVaJa fOUEGf"
  onClick={[Function]}>
  â™¡
</span>
`;
